app/src/main/java/com/bashkevich/androidfundamentals/model/workmanager/MovieLoaderWorker.kt

```diff
@@ -104,6 +112,36 @@ class MovieLoaderWorker(appContext: Context, params: WorkerParameters) : Corouti
                     }
                 }
             }
+
+            val newMovies = database.moviesDao.getAllMovies()
+
+            if (prevMovies.isNotEmpty()) {
+
+                val prevMoviesIds = prevMovies.map { prevMovie -> prevMovie.movieId }
+
+                val newMoviesIds = newMovies.map { newMovie -> newMovie.movieId }
+
+                val moviesIdsToDelete = prevMoviesIds.minus(newMoviesIds)
+
+                moviesIdsToDelete.forEach { movieIdToDelete ->
+                    database.moviesDao.deleteMovieById(movieIdToDelete)
+
+                    database.moviesDao.deleteMovieGenreById(movieIdToDelete)
```

Comment:
Shouldn't it be deleted automatically in cascade?

Reply:
by default not but I  created Foreign keys for that.